### PR TITLE
feat: allow customizing the default prompt in Less pager

### LINKED ISSUE #2033: feat: allow customizing the default prompt in Less pager
## Summary

JLine's built-in `Less` pager displays a hardcoded `:` as the default prompt (the status line at the bottom of the screen when no message is active). GNU `less` supports a `-P` option to customize this prompt, but JLine's `Less` has no equivalent.

## Use case

Applications embedding JLine's pager for end users who may not be familiar with `less` want to show a helpful hint instead of the bare `:`, e.g.:

```
'q' to Quit, <Space> for more, 'h' for Help
```

This is especially relevant when piping content via an `InputStream` — users unfamiliar with pagers see `:` and don't know what to do.

## Proposed change

Add a `defaultPrompt` field to `Less` with a public setter:

```java
protected String defaultPrompt;

public Less defaultPrompt(String prompt) {
    this.defaultPrompt = prompt;
    return this;
}
```

Then in `display()`, replace the hardcoded `":"` fallback:

```java
// Before:
} else {
    msg.append(":");
}

// After:
} else if (defaultPrompt != null) {
    msg.style(AttributedStyle.INVERSE);
    msg.append(defaultPrompt);
    msg.style(AttributedStyle.INVERSE.inverseOff());
} else {
    msg.append(":");
}
```

This is a minimal, backward-compatible change. Users who don't set a `defaultPrompt` see no difference.

## References

- Stack Overflow question: https://stackoverflow.com/questions/79971911
- Current hardcoded prompt: `Less.java` line ~1410 (`msg.append(":");`)

### PR DESCRIPTION
**Pull request by gnodet**
Fixes #2033

## Summary

Adds support for customizing the default prompt (status line) in the `Less` pager:

- Adds a `defaultPrompt` field with a fluent setter (`Less.defaultPrompt(String)`), allowing applications to replace the hardcoded `:` with a custom string (e.g., `'q' to Quit, <Space> for more, 'h' for Help`).
- Adds the `-P` / `--prompt=string` command-line option to set the default prompt from the shell.
- The custom prompt is rendered with inverse video style, consistent with how other messages are displayed.
- Fully backward-compatible: users who do not set a `defaultPrompt` see no change.

## Test plan

- [x] Build succeeds (`./mvx mvn install -DskipTests`)
- [x] Spotless formatting passes
- [x] Existing tests pass (no regressions)

## Summary by CodeRabbit

* **New Features**
  * Added a configurable default prompt for the pager, including `-P` / `--prompt=string` support.
  * Added a fluent option to set the default prompt; it appears when no other status text is active.
* **Bug Fixes**
  * Improved prompt rendering precedence, including correct behavior when a display pattern is set or when an empty prompt is provided.
* **Tests**
  * Added coverage for the new prompt option, chaining behavior, rendering precedence, and edge cases.

### CODE REVIEW

File: builtins/src/test/java/org/jline/builtins/LessTest.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        try (LineDisciplineTerminal terminal = newTerminal(output)) {
+            Less less = newDisplayableLess(terminal);
+            less.defaultPrompt("");
+
+            // An empty (but non-null) prompt is still "set", so it must be handled without
+            // throwing and without falling back to the ":" default rendering path.
+            boolean fitsOneScreen = less.display(false);
+
+            assertFalse(fitsOneScreen);
+        }
```
[the remark this answers is not in the document]
Comment (gnodet) [reply builtins/src/test/java/org/jline/builtins/LessTest.java:224]: Addressed in 9ce0330. The test now strips ANSI escapes and asserts `assertFalse(plainText.contains(":"), "Empty defaultPrompt should not fall back to the ':' default")` to actually verify the rendering behavior.

**Comment by gnodet** [comment]
Both nitpicks from the CodeRabbit review (commit `63a729d`) have been addressed in `9ce0330`:

1. **Weak `:` fallback assertion (lines 149-159)** — the test now strips ANSI escape sequences via `stripAnsi()` before asserting `contains(":")`, so the check matches only the visible prompt character and cannot be satisfied by a stray colon inside an ANSI sequence.

2. **Empty prompt test missing output verification (lines 197-211)** — the test now inspects the ANSI-stripped output and asserts `assertFalse(plainText.contains(":"), ...)` to confirm the empty `defaultPrompt` does not fall back to the `":"` default rendering path.
